ocean_lib/web3_internal/transactions.py

```diff
@@ -30,14 +31,10 @@ def send_ether(from_wallet: Wallet, to_address: str, ether_amount: int):
         "from": from_wallet.address,
         "to": to_address,
         "value": w3.toWei(ether_amount, "ether"),
+        "chainId": w3.eth.chain_id,
     }
-    _ = w3.eth.estimate_gas(tx)
```

Comment:
I am concerned with the removal of the tx assignment, can I ask you why you did it? It seems not connected with chainId at all.

Reply:
It is not connected with `chainId`, you are right, but I thought that it was completely unnecessary, because you try to estimate the gas with `w3.eth.estimate_gas()` of the tx and after that you replace the gas value with 5000 which I don't exactly know where it comes from. The main point, as I discussed with @DMats, was to keep consistency with `estimate_gas` function.

Reply:
@trentmc What do you think? Should we keep the initial version with not using the `estimate_gas` value in the tx dict? Or does it seem fine like this?

Reply:
If you don't understand why something is there, don't remove it. Especially if it's not in the scope of the issue.

Reply:
Ok, I will keep the initial version, I thought it was unnecessary. No problem.

Reply:
Update: I try to revert the tx dict with the gas value set to 500000, but it actually affects the chain id value and by default my unit test for `send_ether` function where it was declared in the first place. After setting the gas value up to 500000, I applied to formula for the `chainId` and it turned out to be negative and I didn't see any negative chain ids in other documentation. In conclusion, `web3.eth.estimate_gas` does affect the scope of the task, so it should be modified as above to actually get a valid chain id.

Reply:
Sending ether from one address to another costs 21000 gas, so arguably we should just set the gas limit to 21000.

That said, I think using the `estimate_gas` function works just as well so I don't think further changes are required. 

According to the docs it "Executes the given transaction locally without creating a new transaction on the blockchain. Returns amount of gas consumed by execution which can be used as a gas estimate."

@mariacarmina I suspect that you got a "negative chain_id" because you accidentally ran `web3.eth.estimate_gas` on a transaction that didn't have `chain_id` in it so it fell back to the old calculation of `v` resulting in `v = 27` or `v = 28`.

Reply:
I added the `chainId` field to the last tx so we can keep the initial form as @trentmc said before. What do you think @DMats?

Reply:
Honestly, I think it should look like one of the following.

```python
tx = {
    "from": from_wallet.address,
    "to": to_address,
    "value": w3.toWei(ether_amount, "ether"),
    "chainId": w3.eth.chain_id,
    "gas": 21000,
}
```

or like this:

```python
tx = {
    "from": from_wallet.address,
    "to": to_address,
    "value": w3.toWei(ether_amount, "ether"),
    "chainId": w3.eth.chain_id,
}
tx["gas"] = web3.eth.estimate_gas(tx)
```

Reply:
I have approached the second option, but @trentmc should I keep the second option that @DMats mentioned?

Reply:
I can use the second option that you have purposed @DMats, because @trentmc agreed with that in the private conversation as far as I see.

Reply:
I am fine with this.